ksqldb-rest-app/src/test/java/io/confluent/ksql/api/server/OldApiUtilsTest.java

```diff
@@ -0,0 +1,97 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.rest.EndpointResponse;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.util.VertxCompletableFuture;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OldApiUtilsTest {
+
+  @Mock
+  private Server server;
+  @Mock
+  private HttpServerRequest request;
+  @Mock
+  private HttpServerResponse response;
+  @Mock
+  private RoutingContext routingContext;
+  @Captor
+  private ArgumentCaptor<Handler<Void>> endHandler;
+
+
+  @Before
+  public void setUp() {
+    when(routingContext.request()).thenReturn(request);
+    when(request.version()).thenReturn(HttpVersion.HTTP_2);
+    when(request.headers()).thenReturn(MultiMap.caseInsensitiveMultiMap());
+    when(request.remoteAddress()).thenReturn(SocketAddress.inetSocketAddress(9000, "remote"));
+    when(response.endHandler(endHandler.capture())).thenReturn(response);
+  }
+
+  @Test
+  public void shouldMaskKsqlRequestQuery() {
+    // Given
+    final String query = "--this is a comment. \n"
+        + "CREATE SOURCE CONNECTOR `test-connector` WITH ("
+        + "    \"connector.class\" = 'PostgresSource', \n"
+        + "    'connection.url' = 'jdbc:postgresql://localhost:5432/my.db',\n"
+        + "    \"mode\"='bulk',\n"
+        + "    \"topic.prefix\"='jdbc-',\n"
+        + "    \"table.whitelist\"='users',\n"
+        + "    \"key\"='username');";
+    final KsqlRequest req = new KsqlRequest(query, ImmutableMap.of(), ImmutableMap.of(), 1L);
+    when(routingContext.getBody()).thenReturn(ServerUtils.serializeObject(req));
+
+    // when, Then
+    final String expected = "CREATE SOURCE CONNECTOR `test-connector` WITH "
+        + "(\"connector.class\"='PostgresSource', "
+        + "'connection.url'='[string]', "
+        + "\"mode\"='[string]', "
+        + "\"topic.prefix\"='[string]', "
+        + "\"table.whitelist\"='[string]', "
+        + "\"key\"='[string]');";
+
+    OldApiUtils.handleOldApiRequest(server, routingContext, KsqlRequest.class, Optional.empty(),
+        (ksqlRequest, securityContext) -> {
+      assertThat(ksqlRequest.getMaskedKsql(), is(expected));
```

Comment:
Does this test actually fail if this assertion fails? I haven't tried running it so if you did then that's definitive -- it just looks like the assertion is contained in an async block which won't actually be executed before the test returns.

Reply:
The assertion actually is in lamda expression instead of async block. Future is returned after this assertion.

Reply:
Ah, good point! I think the newly added `AtomicReference<String> actual` is no longer needed then, right?